src/test/java/teammates/test/cases/webapi/DeleteCourseActionTest.java

```diff
@@ -30,87 +27,66 @@ protected String getRequestMethod() {
 
     @Override
     @Test
-    protected void testExecute() throws Exception {
-        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
-        String instructorId = instructor1OfCourse1.googleId;
-
-        loginAsInstructor(instructorId);
+    public void testExecute() throws Exception {
 
         ______TS("Not enough parameters");
 
         verifyHttpParameterFailure();
 
-        ______TS("Typical case, 2 courses");
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        String instructorId = instructor1OfCourse1.googleId;
+        String courseId = instructor1OfCourse1.courseId;
+
+        ______TS("Typical case, delete a soft-deleted course in Recycle Bin");
 
-        CoursesLogic.inst().createCourseAndInstructor(instructorId, "icdct.tpa.id1", "New course", "UTC");
         String[] submissionParams = new String[] {
-                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+                Const.ParamsNames.COURSE_ID, courseId,
         };
 
-        DeleteCourseAction deleteAction = getAction(submissionParams);
-        JsonResult result = getJsonResult(deleteAction);
-
-        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        loginAsInstructor(instructorId);
+        logic.moveCourseToRecycleBin(courseId);
+        assertEquals(courseId, logic.getSoftDeletedCourseForInstructor(instructor1OfCourse1).getId());
 
-        MessageOutput msg = (MessageOutput) result.getOutput();
-        assertEquals("The course idOfTypicalCourse1 has been deleted. You can restore it from the Recycle Bin manually.",
-                msg.getMessage());
+        DeleteCourseAction deleteCourseAction = getAction(submissionParams);
+        JsonResult result = getJsonResult(deleteCourseAction);
+        MessageOutput message = (MessageOutput) result.getOutput();
 
-        List<CourseAttributes> courseList = CoursesLogic.inst().getCoursesForInstructor(instructorId);
-        assertEquals(1, courseList.size());
-        assertEquals("icdct.tpa.id1", courseList.get(0).getId());
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        assertEquals("The course " + courseId + " has been permanently deleted.", message.getMessage());
+        assertThrows(NullPointerException.class, () -> logic.getSoftDeletedCourseForInstructor(instructor1OfCourse1));
```

Comment:
Rather than catch NPE, you can get the course directly using `logic.getCourse()` ?

Reply:
Sorry, how would this work? `logic.getCourse(instructor1OfCourse1.courseId)` returns null, do I just check that it returns null?

Reply:
Exactly. assertNull()